src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -175,11 +235,83 @@ private SafeCurlHandle CreateRequestHandle(RequestCompletionSource state)
                 Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_FOLLOWLOCATION, 1);
             }
 
-            // TODO: Handle headers, proxy and other options
+            SetProxyOptions(state, requestHandle);
+            // TODO: Handle headers and other options
 
             return requestHandle;
         }
 
+        private static void SetProxyOptions(RequestCompletionSource state, SafeCurlHandle requestHandle)
+        {
+            var requestUri = state.RequestMessage.RequestUri;
+            Debug.Assert(state.Handler != null);
+            if (state.Handler._proxyPolicy == ProxyUsePolicy.DoNotUseProxy)
+            {
+                Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_PROXY, string.Empty);
+                return;
+            }
+
+            if ((state.Handler._proxyPolicy == ProxyUsePolicy.UseDefaultProxy) || (state.Handler.Proxy == null))
+            {
+                return;
+            }
+
+            Debug.Assert( (state.Handler.Proxy != null) && (state.Handler._proxyPolicy == ProxyUsePolicy.UseCustomProxy));
+            if (state.Handler.Proxy.IsBypassed(requestUri))
+            {
+                Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_PROXY, string.Empty);
+                return;
+            }
+
+            var proxyUri = state.Handler.Proxy.GetProxy(requestUri);
+            if (proxyUri == null)
+            {
+                return;
+            }
+
+            Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_PROXYTYPE, CURLProxyType.CURLPROXY_HTTP);
```

Comment:
Thanks.  It's ok to do it in the next PR.  Even if these functions don't fail on such values today, since it's documented that it could fail, it'll be good to have those checks, in case for example in a future update it could fail.